### PR TITLE
fix(protocol): allow resolver to return zero address for EtherVault

### PR DESCRIPTION
**Pull request by dionysuzx**
this is already checked by the addressResolved because we pass in `false` to the function. this makes sure it cannot return the zero address.

### CODE REVIEW

**Comment by cyberhorsey** [comment]
I think this should actually be `true` for allowing the 0 address, as we only deploy an EtherVault on Layer 2, not Layer 1?

**Comment by davidtaikocha** [comment]
> I think this should actually be `true` for allowing the 0 address, as we only deploy an EtherVault on Layer 2, not Layer 1?

I think so.

File: packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol [hunk 1]
```diff
@@ -78,11 +78,10 @@ library LibBridgeProcess {
 
         // We retrieve the necessary ether from EtherVault
         address ethVault = resolver.resolve("ether_vault", false);
-        if (ethVault != address(0)) {
-            EtherVault(payable(ethVault)).releaseEther(
-                message.depositValue + message.callValue + message.processingFee
-            );
-        }
+        EtherVault(payable(ethVault)).releaseEther(
```
Comment (dantaik) [inline packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol:81]: Good catch, but I think we should have the above change as we do allow the ethVault to be address(0)

File: packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol [hunk 2]
```diff
@@ -78,11 +78,10 @@ library LibBridgeProcess {
 
         // We retrieve the necessary ether from EtherVault
         address ethVault = resolver.resolve("ether_vault", false);
```
Comment (dantaik) [inline packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol:80]:
```suggestion
        address ethVault = resolver.resolve("ether_vault", true);
```

**Review by dantaik — CHANGES_REQUESTED** [review]

**Comment by d1onys1us** [comment]
@dantaik i applied the suggested changes, thanks. but i'm curious, why wasn't this reverting then? it should've been throwing errors when withdrawing on layer 1 right?

**Review by davidtaikocha — APPROVED** [review]

**Comment by davidtaikocha** [comment]
> @dantaik i applied the suggested changes, thanks. but i'm curious, why wasn't this reverting then? it should've been throwing errors when withdrawing on layer 1 right?

I believe its because the `bool allowZeroAddress` parameter of `AddressReslover.reslove()` function is introduced in [a PR (signal service)](https://github.com/taikoxyz/taiko-mono/commit/53adc983d2cd1fd6d929a06273f5341aacd17645#diff-af382ffb911c4e17a4797dcc2df96a9fde2272b480ab6067b607e4aac5df75b3R58) which is merged after the Alpha-1 release.

**Review by dantaik — APPROVED** [review]

**Pull request merged**